File: Kotlin-Dagger2/src/main/kotlin/org/codecop/lcdnumbers/CommandLineArguments.kt
package org.codecop.lcdnumbers

class CommandLineArguments(private val args: Array<String>) {
    fun isHelpRequired() = args.isEmpty()
    
    val numberToDisplay: Int
        get() = args[0].toInt()
    
    val scaling: Scaling
        get() = if (args.size > 1) Scaling.of(args[1].toInt()) else Scaling.NONE
}

File: Csharp-NAutowired/LcdNumbers/CommandLineArguments.cs
using System;
using NAutowired.Core.Attributes;

namespace Org.Codecop.Lcdnumbers
{
    public class CommandLineArguments : ICommandLineArguments
    {
        private readonly string[] args;

        public CommandLineArguments(string[] args)
        {
            this.args = args;
        }

        public bool IsHelpRequired()
        {
            return args.Length == 0;
        }

        public int GetNumberToDisplay()
        {
            return Convert.ToInt32(args[0]);
        }

        public Scaling GetScaling()
        {
            return args.Length > 1 ? Scaling.Of(Convert.ToInt32(args[1])) : Scaling.None;
        }
    }
}
